x-pack/legacy/plugins/maps/public/layers/styles/vector/properties/dynamic_color_property.test.js

```diff
@@ -92,91 +99,121 @@ const defaultLegendParams = {
   isLinesOnly: false,
 };
 
-test('Should render ordinal legend', async () => {
-  const colorStyle = makeProperty({
-    color: 'Blues',
-    type: undefined,
-  });
+describe('ordinal', () => {
+  test('Should render ordinal legend as bands', async () => {
+    const colorStyle = makeProperty({
+      color: 'Blues',
+      type: undefined,
+    });
 
-  const legendRow = colorStyle.renderLegendDetailRow(defaultLegendParams);
+    const legendRow = colorStyle.renderLegendDetailRow(defaultLegendParams);
 
-  const component = shallow(legendRow);
+    const component = shallow(legendRow);
 
-  expect(component).toMatchSnapshot();
-});
+    // Ensure all promises resolve
```

Comment:
Why all the async logic in the tests? The render methods are all sync.

Reply:
due to async fetching of label display.